File: src/main/java/tictactoe/CompletedGamesAnalysis.java
package tictactoe;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;

public class CompletedGamesAnalysis {
    public static  CompletedGamesReport generateCompletedGamesReport(InputStream inputStream) {
        int numberOfOWinner = 0;
        int numberOfXWinner = 0;
        int numberOfDraw = 0;

        try (BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream))) {
            String line;
            while ((line = reader.readLine()) != null) {
                if (!line.isEmpty()){
                    Player winner = new BoardState(line).getWinner();
                    if (winner == Player.O) {
                        numberOfOWinner++;
                    } else if (winner == Player.X) {
                        numberOfXWinner++;
                    } else {
                        numberOfDraw++;
                    }
                }
            }
        } catch (IOException e) {
            e.printStackTrace();
        }

        return new CompletedGamesReport(numberOfXWinner, numberOfOWinner, numberOfDraw);
    }
}

File: src/main/kotlin/tictactoe/CompletedGamesAnalysis.kt
package tictactoe

import java.io.InputStream

/**
 * static method to generate completed game report
 *
 * @return completed games report
 */
fun generateCompletedGamesReport(inputStream: InputStream): CompletedGamesReport {
    var numberOfOWinner = 0
    var numberOfXWinner = 0
    var numberOfDraw = 0
    inputStream.bufferedReader().forEachLine { line ->
        if (line.isNotEmpty()){
            when(BoardState(line).getWinner()) {
                Player.O -> numberOfOWinner++
                Player.X -> numberOfXWinner++
                else -> numberOfDraw++
            }
        }
    }

    return CompletedGamesReport(
        numberOfXWinner, numberOfOWinner, numberOfDraw
    )
}

fun main() {
    val inputStream = object {}.javaClass.getResourceAsStream("/completed_games.txt")
    val result = generateCompletedGamesReport(inputStream)
    println(result.getContent())
}
